stock_picking_product_kit_helper/models/stock_picking.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2019 Kitti U. - Ecosoft <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    product_kit_helper_ids = fields.One2many(
+        comodel_name='stock.picking.product.kit.helper',
+        string='Product Kit Helper Lines',
+        inverse_name='picking_id',
+        readonly=False,
+        states={'done': [('readonly', True)], 'cancel': [('readonly', True)]},
+    )
+    has_product_kit = fields.Boolean(
+        string='Has Product Kit',
+        compute='_compute_has_product_kit',
+        help="True if there is at least 1 product kit in the sales order",
+    )
+
+    @api.model
+    def _is_product_kit(self, product, company):
+        BOM = self.env['mrp.bom'].sudo()
+        bom = BOM._bom_find(product=product,
+                            company_id=company.id)
+        return bom and bom.type == 'phantom'
+
+    @api.multi
+    def _compute_has_product_kit(self):
+        for picking in self:
+            if any(self._is_product_kit(line.product_id, line.company_id)
+                   for line in picking.move_lines.mapped('sale_line_id')):
+                picking.has_product_kit = True
+
+    @api.multi
+    def show_product_kit(self):
+        """Find move_lines with product kit to create helper line."""
+        self.ensure_one()
+        BOM = self.env['mrp.bom'].sudo()
+        helpers = []
+        for sale_line in self.move_lines.mapped('sale_line_id'):
+            bom = BOM._bom_find(product=sale_line.product_id,
+                                company_id=sale_line.company_id.id)
+            if bom and bom.type == 'phantom':  # Create product kit line
+                helpers.append((0, 0, {'sale_line_id': sale_line.id,
+                                       'product_id': sale_line.product_id.id,
+                                       'product_uom_qty': 0.0,
+                                       }))
+        self.product_kit_helper_ids.unlink()
+        self.write({'product_kit_helper_ids': helpers})
+
+    @api.multi
+    def action_product_kit_helper(self):
+        """Assign product kit's quantity to stock move."""
+        self.ensure_one()
+        if self.state in ('done', 'cancel'):
+            raise ValidationError(
+                _('Product Kit Helper is not allowed on current state'))
+        for helper in self.product_kit_helper_ids:
+            helper.action_explode_helper()
+
+
+class StockPickingProductKitHelper(models.Model):
```

Comment:
Shouldn't be a Wizard instead ?

Reply:
@rousseldenis we have think through this point also. But functional point of view, @wipawii confirm that, it is more user friendly + keeping rigid history on what user is doing.

Reply:
@wipawii, another way is to make it as wizard, and write log in chatter.

Reply:
As you want. It was an intuitive remark.